src/graph/AdminJobExecutor.cpp

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "http/HttpClient.h"
+#include "graph/AdminJobExecutor.h"
+#include "process/ProcessUtils.h"
+#include "webservice/Common.h"
+
+#include <folly/executors/Async.h>
+#include <folly/futures/Future.h>
+
+namespace nebula {
+namespace graph {
+
+AdminJobExecutor::AdminJobExecutor(Sentence *sentence,
+        ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<AdminSentence*>(sentence);
+}
+
+void AdminJobExecutor::execute() {
+    LOG(INFO) << __func__ << " enter";
+    auto opEnum = toAdminJobOp(sentence_->getType());
+
+    if (opNeedsSpace(opEnum)) {
+        auto status = checkIfGraphSpaceChosen();
+        if (!status.ok()) {
+            DCHECK(onError_);
+            onError_(std::move(status));
+            return;
+        }
+        std::string spaceName = ectx()->rctx()->session()->spaceName();
+        sentence_->addPara(spaceName);
```

Comment:
here is a little confusing. Why?

Reply:
Just want to treat space name as one parameter. But some job has this, others are not.

Reply:
A little suggestion, this parameter set during the AST phase only. Here we can create a private variable `spaceName_` ，WDTY？

Reply:
Actually, I think there should be some thing like ExecutionPlan between sentence and executor. As there isn't currently, I think modify AST directly is not that harmful.

Reply:
In one hand, I thought there should sth like ExecutionPlan between sentence and executor. As there isn't, it is the only way to change AST.

Reply:
But on the other hand, I think your suggestion make sense. Let the sentence be a pure AST. Use a local vector to hold all these paras, thanks.